### ISSUE TITLE
[bug]  Name containing '-' issues a discouraged warning

### ISSUE THREAD

**Issue by offa** [conan-io/conan#13412]
### Environment details

* Operating System+version: Linux
* Compiler+version: GCC 12
* Conan version: 2.0.1
* Python version: 3.10

### Steps to reproduce

1. Create a conanfile.py with a name attribute containing a `-` (eg. `name = "influxdb-cxx"`)
2. Do `conan create`

```sh
conan new cmake_lib -d name=warning-repro -d version=0.1
conan create .
```

### Logs

```
conan create .

======== Exporting recipe to the cache ========
WARN: Name containing special chars is discouraged 'warning-repro'
[…]
```

**Comment by memsharded** [comment]
Hi @offa 

Thanks for your report.

This is the reason why ``-`` doesn't raise an error, because it is possible to use it.
However, it is so problematic for many build systems and tools that it is warning, to discourage its use. It is very recommended not to use such names, but avoid the ``-`` completely. It wasn't fully forbidden to allow an easier transition and upgrade to 2.0, but the warning is legit, this wouldn't be a bug.

**Comment by offa** [comment]
Thanks, that's fine to me and I'm replace it with `_`.

It should be mention in the docs though, especially if it may cause trouble.

**Comment by memsharded** [comment]
> It should be mention in the docs though, especially if it may cause trouble.

Definitely. But most people won't read the docs, specially if there is not an issue. So the warning in the output is the thing that works way better than the docs.
Lets move this to the docs repo for improvement, thanks for the feedback!
